Rubberduck.Parsing/VBA/IModuleToModuleReferenceManager.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Rubberduck.VBEditor;
+using System.Collections.Generic;
+
+
+namespace Rubberduck.Parsing.VBA
+{
+    public interface IModuleToModuleReferenceManager
+    {
+        void AddModuleToModuleReference(QualifiedModuleName referencingModule, QualifiedModuleName referencedModule);
+        void RemoveModuleToModuleReference(QualifiedModuleName referencedModule, QualifiedModuleName referencingModule);
+        void ClearModuleToModuleReferencesFromModule(QualifiedModuleName referencingModule);
+        void ClearModuleToModuleReferencesFromModule(IEnumerable<QualifiedModuleName> referencingModules);
+        void ClearModuleToModuleReferencesToModule(QualifiedModuleName referencedModule);
+        void ClearModuleToModuleReferencesToModule(IEnumerable<QualifiedModuleName> referencedModules);
+
+        ICollection<QualifiedModuleName> ModulesReferencedBy(QualifiedModuleName referencingModule);
+        ICollection<QualifiedModuleName> ModulesReferencedByAny(IEnumerable<QualifiedModuleName> referencingModules);
+        ICollection<QualifiedModuleName> ModulesReferencing(QualifiedModuleName referencedModule);
+        ICollection<QualifiedModuleName> ModulesReferencingAny(IEnumerable<QualifiedModuleName> referencedModules);
```

Comment:
Totally makes sense. I will switch to readonly-collections. (I had completely forgotten that they exist.)